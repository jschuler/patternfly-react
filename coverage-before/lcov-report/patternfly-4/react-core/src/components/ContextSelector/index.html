<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for patternfly-4/react-core/src/components/ContextSelector</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> patternfly-4/react-core/src/components/ContextSelector
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80.12% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>137/171</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">46.03% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>29/63</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">56.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>17/30</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">79.26% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>107/135</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="ContextSelector.tsx"><a href="ContextSelector.tsx.html">ContextSelector.tsx</a></td>
	<td data-value="87.23" class="pic high"><div class="chart"><div class="cover-fill" style="width: 87%;"></div><div class="cover-empty" style="width:13%;"></div></div></td>
	<td data-value="87.23" class="pct high">87.23%</td>
	<td data-value="47" class="abs high">41/47</td>
	<td data-value="76.92" class="pct medium">76.92%</td>
	<td data-value="13" class="abs medium">10/13</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="86.36" class="pct high">86.36%</td>
	<td data-value="44" class="abs high">38/44</td>
	</tr>

<tr>
	<td class="file high" data-value="ContextSelectorItem.tsx"><a href="ContextSelectorItem.tsx.html">ContextSelectorItem.tsx</a></td>
	<td data-value="96.77" class="pic high"><div class="chart"><div class="cover-fill" style="width: 96%;"></div><div class="cover-empty" style="width:4%;"></div></div></td>
	<td data-value="96.77" class="pct high">96.77%</td>
	<td data-value="31" class="abs high">30/31</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="9" class="abs medium">7/9</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	</tr>

<tr>
	<td class="file high" data-value="ContextSelectorMenuList.tsx"><a href="ContextSelectorMenuList.tsx.html">ContextSelectorMenuList.tsx</a></td>
	<td data-value="95.65" class="pic high"><div class="chart"><div class="cover-fill" style="width: 95%;"></div><div class="cover-empty" style="width:5%;"></div></div></td>
	<td data-value="95.65" class="pct high">95.65%</td>
	<td data-value="23" class="abs high">22/23</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="15" class="abs high">14/15</td>
	</tr>

<tr>
	<td class="file medium" data-value="ContextSelectorToggle.tsx"><a href="ContextSelectorToggle.tsx.html">ContextSelectorToggle.tsx</a></td>
	<td data-value="61.19" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 61%;"></div><div class="cover-empty" style="width:39%;"></div></div></td>
	<td data-value="61.19" class="pct medium">61.19%</td>
	<td data-value="67" class="abs medium">41/67</td>
	<td data-value="23.68" class="pct low">23.68%</td>
	<td data-value="38" class="abs low">9/38</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="63.64" class="pct medium">63.64%</td>
	<td data-value="55" class="abs medium">35/55</td>
	</tr>

<tr>
	<td class="file high" data-value="contextSelectorConstants.tsx"><a href="contextSelectorConstants.tsx.html">contextSelectorConstants.tsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jan 17 2020 19:28:45 GMT-0500 (Eastern Standard Time)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
<script src="../../../../../block-navigation.js"></script>
</body>
</html>
