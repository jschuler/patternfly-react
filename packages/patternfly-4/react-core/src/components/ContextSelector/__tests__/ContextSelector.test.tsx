// Auto-generated do not edit

import * as React from 'react';
import { mount } from 'enzyme';
import { ContextSelector } from '../ContextSelector';

describe('ContextSelector test', () => {
  it('ContextSelector should match snapshot', () => {
    const view = mount(
      <ContextSelector
        children={<>ReactNode</>}
				className={"''"}
				isOpen={false}
				onToggle={() => undefined as any}
				onSelect={() => undefined as any}
				screenReaderLabel={"''"}
				toggleText={"''"}
				searchButtonAriaLabel={"'Search menu items'"}
				searchInputValue={"''"}
				onSearchInputChange={() => undefined as any}
				searchInputPlaceholder={"'Search'"}
				onSearchButtonClick={() => undefined as any}
      />);
    expect(view).toMatchSnapshot();
  });
});
