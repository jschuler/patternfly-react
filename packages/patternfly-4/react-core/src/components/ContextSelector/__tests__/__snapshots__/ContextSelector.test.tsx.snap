// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextSelector test ContextSelector should match snapshot 1`] = `
<ContextSelector
  className="''"
  isOpen={false}
  onSearchButtonClick={[Function]}
  onSearchInputChange={[Function]}
  onSelect={[Function]}
  onToggle={[Function]}
  screenReaderLabel="''"
  searchButtonAriaLabel="'Search menu items'"
  searchInputPlaceholder="'Search'"
  searchInputValue="''"
  toggleText="''"
>
  <div
    className="pf-c-context-selector ''"
  >
    <span
      hidden={true}
      id="pf-context-selector-label-id-0"
    >
      ''
    </span>
    <ContextSelectorToggle
      aria-labelledby="pf-context-selector-label-id-0 pf-context-selector-toggle-id-0"
      className=""
      id="pf-context-selector-toggle-id-0"
      isActive={false}
      isFocused={false}
      isHovered={false}
      isOpen={false}
      onEnter={[Function]}
      onToggle={[Function]}
      parentRef={null}
      toggleText="''"
    >
      <button
        aria-expanded={false}
        aria-labelledby="pf-context-selector-label-id-0 pf-context-selector-toggle-id-0"
        className="pf-c-context-selector__toggle"
        id="pf-context-selector-toggle-id-0"
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <span
          className="pf-c-context-selector__toggle-text"
        >
          ''
        </span>
        <CaretDownIcon
          aria-hidden={true}
          className="pf-c-context-selector__toggle-icon"
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
          title={null}
        >
          <svg
            aria-hidden={true}
            aria-labelledby={null}
            className="pf-c-context-selector__toggle-icon"
            fill="currentColor"
            height="1em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 320 512"
            width="1em"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              transform=""
            />
          </svg>
        </CaretDownIcon>
      </button>
    </ContextSelectorToggle>
  </div>
</ContextSelector>
`;
